#!/usr/bin/env python
import math
import os.path
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("invert_direction", bool_t, 0, "", default=False)
gen.add("position_offset", double_t, 0, "", default=0.0, min=0.0)
gen.add("process_noise_p", double_t, 0, "", default=0.001, min=0.0)
gen.add("process_noise_v", double_t, 0, "", default=1.0, min=0.0)
gen.add("measurement_noise", double_t, 0, "", default=0.01, min=0.0)
gen.add("mockup_max_current", double_t, 0, "", default=1.0, min=0.0)
gen.add("mockup_current_to_acceleration", double_t, 0, "", default=100.0, min=0.0)

package_name = os.path.basename(os.path.dirname(os.path.dirname(__file__)))
node_name = package_name + "_node"
cfg_name = os.path.splitext(os.path.basename(__file__))[0]
exit(gen.generate(package_name, node_name, cfg_name))
